### PR TITLE
#3466 Make necessary adjustments to definitions API in order to meet …

### PR DESCRIPTION
**Pull request by wu87988622**
…the UI requirements

### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3466 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/SettingDialog.js [hunk 1]
```diff
@@ -70,10 +70,46 @@ const StyleMuiDialogActions = styled(MuiDialogActions)(
...
+        padding-left: ${theme.spacing(3)}px;
+      }
+
+      ::before {
+        content: '';
+        left: ${theme.spacing(3)}px;
+        top: 0;
+        bottom: 0;
+        position: absolute;
+        width: ${theme.spacing(0.25)}px;
```
Comment (saivirtue) [inline ohara-manager/client/src/components/Pipeline/SettingDialog.js]: 蠻特別的用法，這樣其實可以組合出所有想要的數字了

File: ohara-manager/client/src/components/common/Definitions/PositiveDouble.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  const {
+    input: { name, onChange, value },
+    meta = {},
+    helperText,
+    refs,
+    ...rest
+  } = omit(props, ['tableKeys']);
+
+  const hasError =
+    (meta.error && meta.touched) || (meta.error && meta.dirty) ? true : false;
```
Comment (eechih) [inline ohara-manager/client/src/components/common/Definitions/PositiveDouble.js]: (meta.error && meta.touched) || (meta.error && meta.dirty) 本身就可以傳回 true or false。
可簡化成 const hasError = (meta.error && meta.touched) || (meta.error && meta.dirty)

File: ohara-manager/client/src/components/common/Definitions/RenderDefinitions.js [hunk 3]
```diff
@@ -14,132 +14,60 @@
...
-import Port from './Port';
-import Reference from './Reference';
-import StringDef from './StringDef';
-import Table from './Table';
-import Tags from './Tags';
-import PositiveInt from './PositiveInt';
-import ClassDef from './ClassDef';
-import Long from './Long';
+import { Form } from 'react-final-form';
+import RednerDefinition from './RenderDefinition';
```
Comment (eechih) [inline ohara-manager/client/src/components/common/Definitions/RenderDefinitions.js]: Typo: RednerDefinition

File: ohara-manager/client/src/components/common/Definitions/RenderDefinition.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+import ObjectKey from './ObjectKey';
+import ObjectKeys from './ObjectKeys';
+import PositiveDouble from './PositiveDouble';
+import PositiveLong from './PositiveLong';
+import PositiveShort from './PositiveShort';
+
+const RenderDefinition = props => {
+  const { def, topics, files } = props;
+
+  const ref = React.createRef();
```
Comment (eechih) [inline ohara-manager/client/src/components/common/Definitions/RenderDefinition.js]: RenderDefinition 不應該強迫自己建立一個 Ref，應該由外部決定是否需要 Ref。
建議由 props 傳入 ref

File: ohara-manager/client/src/components/common/Definitions/PositiveDouble.js (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/common/Definitions/PositiveDouble.js]: nice

File: ohara-manager/client/src/components/common/Definitions/RenderDefinitions.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/common/Definitions/RenderDefinitions.js]: thanks

File: ohara-manager/client/src/components/common/Definitions/RenderDefinition.js (hunk 4 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/common/Definitions/RenderDefinition.js]: ok thanks

**Pull request merged**